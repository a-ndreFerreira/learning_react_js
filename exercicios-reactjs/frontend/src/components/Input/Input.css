fieldset {
    max-width: 21.87rem;
    margin: 0 auto;
    padding: .5rem;
    border-radius: .5rem;
    border: none;
}

/* fieldset input:focus {
    border: none;
} */

fieldset input {
    width: 100%;
    padding: .4rem .7rem;
    border-radius: .9rem;
}

@media(min-width: 46rem) {
    fieldset {
        max-width: 30rem;
    }

    /* fieldset input {
        width: 100%;
    } */
}