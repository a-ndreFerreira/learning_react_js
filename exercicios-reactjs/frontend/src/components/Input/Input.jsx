import './Input.css'

const Input = ({ onChange, searchValue, placeholder }) => {
    return (
        <fieldset>
            <legend>Buscar</legend>
            <input type='search'
                name='search'
                id='search'
                onChange={onChange}
                value={searchValue}
                placeholder={placeholder}
            />
        </fieldset>
    )
}

export default Input