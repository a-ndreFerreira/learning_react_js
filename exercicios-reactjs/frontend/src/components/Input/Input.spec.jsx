import { describe, expect, vi } from "vitest";
import { render, screen, within } from "@testing-library/react";
import userEvent from '@testing-library/user-event'
import Input from "./Input";

describe('<Input />', () => {
    it('should be render all the props an component INPUT in screen', () => {
        const fn = vi.fn();
        render(<Input onChange={fn} searchValue='title' placeholder='placeholder' />);

        expect(within(screen.getByRole('group')).getByText('Buscar')).toBeInTheDocument();

    })

    it('should be value input in the searchValue', () => {
        const fn = vi.fn();
        render(<Input onChange={fn} searchValue='title' placeholder='placeholder' />);

        const input = screen.getByPlaceholderText('placeholder');

        expect(input.value).toBe('title');
    })

    it('should call onChange on each key pressed', async () => {
        const fn = vi.fn();
        render(<Input onChange={fn} placeholder='placeholder' />);

        const input = screen.getByPlaceholderText('placeholder');

        const value = 'o valor';

        await userEvent.type(input, value);

        expect(input.value).toBe(value);
        expect(fn).toHaveBeenCalledTimes(value.length);

    })

    it('should match snapshot', () => {
        const { container } = render(<Input />);

        expect(container).toMatchSnapshot();
    })

})