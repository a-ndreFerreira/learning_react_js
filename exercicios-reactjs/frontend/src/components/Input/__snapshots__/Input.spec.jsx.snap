// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Input /> > should match snapshot 1`] = `
<div>
  <fieldset>
    <legend>
      Buscar
    </legend>
    <input
      id="search"
      name="search"
      type="search"
    />
  </fieldset>
</div>
`;
